### PR TITLE
Implemented required functionality

### PR DESCRIPTION
**Pull request by andrei-tyk**
## Description

Tyk Helm Chart currently sets the same port from the values.yaml to port and  targetPort.  This leads to a situation in which configuring a load balancer on different ports (e.g. 80 and 443) is impossible.

The change in the PR should allow for different values to be set for port and targetPort on the Tyk Gateway.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by komalsukhani — APPROVED** [review]

**Comment by komalsukhani** [comment]
@andrei-tyk Can you update the description?

**Comment by andrei-tyk** [comment]
> The new `containerPorts` field is not introduced in `tyk-oss/values.yaml` file yet. Can you please update PR accordingly?

Adapted

File: tyk-mdcb-data-plane/values.yaml [hunk 1]
```diff
@@ -249,8 +252,8 @@ tyk-gateway:
       tags: ""
 
     # used to decide whether to send the results back directly to Tyk without a hybrid pump
-    # if you want to send analytics to control plane instead of pump, change value to "rpc"
-    analyticsConfigType: ""
+    # if you want to use a hybrid pump, please change "rpc" to ""
+    analyticsConfigType: "rpc"
```
Comment (buraksekili) [inline tyk-mdcb-data-plane/values.yaml:256]: i guess we do not need to change this. what do you think? can you please double-check?

File: tyk-mdcb-data-plane/values.yaml [hunk 2]
```diff
@@ -310,9 +313,6 @@ tyk-pump:
       # image pull policy
       pullPolicy: IfNotPresent
 
-    # image pull secrets to use when pulling images from repository
-    imagePullSecrets: []
```
Comment (buraksekili) [inline tyk-mdcb-data-plane/values.yaml:314]: do we need to delete this section?

Comment (andrei-tyk) [reply tyk-mdcb-data-plane/values.yaml:327]: readded

File: tyk-mdcb-data-plane/values.yaml (hunk 1 above)
Comment (andrei-tyk) [reply tyk-mdcb-data-plane/values.yaml]: it was due to a merge conflict, fixed

File: tyk-mdcb-data-plane/values.yaml [hunk 3]
```diff
@@ -323,8 +326,8 @@ tyk-pump:
...
+      # image pull secrets to use when pulling images from repository
+      imagePullSecrets: []
```
Comment (buraksekili) [inline tyk-mdcb-data-plane/values.yaml:330]:
```suggestion
    # image pull secrets to use when pulling images from repository
    imagePullSecrets: []
```

**Review by buraksekili — COMMENTED** [review]
please check the following change. other than that, it looks good.

**Review by buraksekili — APPROVED** [review]

**Pull request merged**